cli/module_generate/scripts/generate_stubs.py

```diff
@@ -3,15 +3,97 @@
 import subprocess
 import sys
 from importlib import import_module
+from typing import List
 
 
-def return_attribute(resource_name: str, attr: str) -> ast.Attribute:
+def get_final_imports(imports: List[str]) -> List[str]:
+    final_imports = []
+    for i in imports:
+        if i not in final_imports:
+            final_imports.append(i)
+    return final_imports
+
+
+def return_attribute(value: str, attr: str) -> ast.Attribute:
     return ast.Attribute(
-        value=ast.Name(id=resource_name, ctx=ast.Load()),
+        value=ast.Name(id=value, ctx=ast.Load()),
         attr=attr,
         ctx=ast.Load())
 
 
+def update_annotation(
+    resource_name: str,
+    annotation: ast.AST,
+    nodes: List[str],
+    parent: str
+) -> ast.AST:
+    if isinstance(annotation, ast.Name) and annotation.id in nodes:
+        value = f"{resource_name}.{parent}" if parent else resource_name
+        return return_attribute(value, annotation.id)
+    elif isinstance(annotation, ast.Subscript):
+        annotation.slice = update_annotation(
+            resource_name,
+            annotation.slice,
+            nodes,
+            parent)
+        return annotation
+    return annotation
+
+
+def replace_async_func(
+    resource_name: str,
+    func: ast.AsyncFunctionDef,
+    nodes: List[str],
+    parent: str = ""
+) -> None:
+    for arg in func.args.args:
+        arg.annotation = update_annotation(
+            resource_name,
+            arg.annotation,
+            nodes,
+            parent)
+    func.body = [
+        ast.Raise(
+            exc=ast.Call(func=ast.Name(id='NotImplementedError',
+                                       ctx=ast.Load()),
+                         args=[], 
+                         keywords=[]),
+            cause=None)
+    ]
+    func.decorator_list = []
+    if isinstance(func.returns, (ast.Name, ast.Subscript)):
+        func.returns = update_annotation(
+            resource_name, func.returns, nodes, parent
+        )
+
+
+def parse_subclass(
+        resource_name: str, stmt: ast.ClassDef, base=""
+) -> List[str]:
+    nodes, nodes_to_remove = [], []
+    base = base if base else resource_name
+    stmt.bases = [ast.Name(id=f"{base}.{stmt.name}", ctx=ast.Load())]
+    for cstmt in stmt.body:
+        if isinstance(cstmt, ast.Expr) or (
+            isinstance(cstmt, ast.FunctionDef) and cstmt.name == "__init__"
+        ):
+            nodes_to_remove.append(cstmt)
+        elif isinstance(cstmt, ast.AnnAssign):
+            nodes.append(cstmt.target.id)
+            nodes_to_remove.append(cstmt)
+        elif isinstance(cstmt, ast.ClassDef):
+            parse_subclass(resource_name, cstmt, stmt.name, stmt.bases[0].id)
```

Comment:
Thanks for catching this! The mismatch on arguments is a relic of some changes I made, so I'll fix that.

This was supposed to handle the case if we had a nested class in a subclass. We do need the whole `parse_subclass` function outside of the return because all the changes will be reflected in that final return `ast.unparse(stmt)` call. That's why it doesn't need the return. I guess I can move reformat this entire function so that everything is in a helper function but the return.